### DISCUSSION TITLE
Query options from Meson.

### DISCUSSION THREAD

**Discussion by sechshelme** [mesonbuild/meson#13309] [General]
Query options from Meson.

I have the following example, it has options like `--strip` or `-Db_ndebug=true`
```
meson setup --cross-file ../win64cross.tmp --buildtype=debugoptimized --strip -Db_ndebug=true ../freetype/
```

Is there a command like `--help` that spits out all these options?

**Comment by eli-schwartz** [comment]
The equivalent of `./configure --help` for meson is `meson configure`. Used with a source directory it lists available options. Used with a build directory it lists the current values as well.

**Comment by sechshelme** [comment]
I just noticed that there are differences when calling Meson --help.
`meson --help` and `meson ../freetype/ --help` spit out different information.
If you now look at the -D parameter, you only get the following: `Set the value of an option, can be used several times to set multiple options.`
In my example above it says `-Db_ndebug=true`.
How do I get this kind of information, what is possible with it?
Is it project specific?
